src/trace-plugin-interface.js

```diff
@@ -0,0 +1,244 @@
+ * Adds a label to the underlying span.
+ * @param {string} key The name of the label to add.
+ * @param {*} value The value of the label to add.
+ */
+ChildSpan.prototype.addLabel = function(key, value) {
+  this.span_.addLabel(key, value);
+};
+
+/**
+ * Ends the underlying span. This function should only be called once.
+ */
+ChildSpan.prototype.endSpan = function() {
+  this.span_.close();
+};
+
+/**
+ * An object that is associated with a single root span. It exposes
+ * functions for adding labels to or closing the associated span.
+ */
+function Transaction(agent, context) {
+  this.agent_ = agent;
+  this.context_ = context;
+}
+
+/**
+ * Adds a label to the underlying span.
+ * @param {string} key The name of the label to add.
+ * @param {*} value The value of the label to add.
+ */
+Transaction.prototype.addLabel = function(key, value) {
+  this.context_.addLabel(key, value);
+};
+
+/**
+ * Ends the underlying span. This function should only be called once.
+ */
+Transaction.prototype.endSpan = function() {
+  this.context_.close();
+};
+
+/**
+ * Runs the given function in a child span, passing it an object that
+ * exposes an interface for adding labels and closing the span.
+ * @param {object} options An object that specifies options for how the child
+ * span is created and propogated.
+ * @param {string} options.name The name to apply to the child span.
+ * @param {?function(string, string)} options.setHeader A function describing
+ * how to set a header field with the given field name to a given value,
+ * respectively. If supplied, it will be called to set the header field in an
+ * outgoing request associated with this child span to the field name
+ * 'x-cloud-trace-context'.
+ * @param {?number} options.skipFrames The number of stack frames to skip when
+ * collecting call stack information for the root span, starting from the top;
+ * this should be set to avoid including frames in the PluginAPI. Defaults to 0.
+ * @param {function(ChildSpan)} fn A function that will be called exactly
+ * once, with a ChildSpan object exposing an interface operating on the child
+ * span.
+ * @returns The return value of calling fn.
+ */
+Transaction.prototype.runInChildSpan = function(options, fn) {
+  options = options || {};
+  var childContext = this.agent_.startSpan(options.name, {},
+    options.skipFrames ? options.skipFrames + 1 : 1);
+  // If the options object passed in has the setHeader field set,
+  // use it to set trace metadata in an outgoing request.
+  if (is.fn(options.setHeader)) {
+    options.setHeader('x-cloud-trace-context',
+      this.agent.generateTraceContext(childContext, true));
+  }
+  return fn(new ChildSpan(this.agent_, childContext));
+};
+
+/**
+ * PluginAPI constructor. Don't call directly - a PluginAPI object will be passed to
+ * PluginAPIs themselves
+ * TODO(kjin): Should be called something else
+ */
+function PluginAPI(agent) {
+  this.agent = agent;
+  this.logger = agent.logger;
+  this.config = {
```

Comment:
Done - exposed `enhancedReportingEnabled()` on API that returns the value of `agent.config.enhancedDatabaseReporting`.